shared/modules/doc/README.md

```diff
@@ -63,11 +76,19 @@ if err != nil {
 }
 ```
 
+For an example of a module that uses a `ModuleOption`, you can search for `WithCustomRPCURL` within the codebase. The code might have changed since this document was written so we are referring to the commit hash [19bf4d3f](https://github.com/pokt-network/pocket/tree/19bf4d3f6507f5d406d9fafdb69b81359bccf110).
+
+Essentially the `ModuleOption` sets a custom RPC URL for the module at runtime.
+
+
 #### Interacting & Registering with the `bus`
 
 The `bus` is the specific integration mechanism that enables the greater application.
 
-When a module is constructed via the `Create(bus modules.Bus)` function, it is expected to internally call `bus.RegisterModule(module)`, which registers the module with the `bus` so its sibling modules can access it synchronously via a DI-like pattern.
+When a module is constructed via the `Create(bus modules.Bus, options ...modules.ModuleOption)` function, it is expected to internally call `bus.RegisterModule(module)`, which registers the module with the `bus` so its sibling modules can access it synchronously via a DI-like pattern.
+
+Probably worth mentioning that we implemented a `ModulesRegistry` module that takes care of the module registration and retrieval. This module is registered with the `bus` at the application level, it is accessible to all modules via the `bus` interface and it's also mockable as you would expect.
```

Comment:
This seems like something that should be represented in the table of contents. Wdyt about 1) giving this paragraph its own heading (e.g. "Module registry" or something) and 2) maybe linking to or embedding an example of registration and/or retrieval?

Reply:
Totally, I have updated accordingly the TOC and added some color about the modules registry

https://github.com/pokt-network/pocket/pull/623/commits/12a71fa6ab56a6158133405d5759311f11a51455

Reply:
Amazing -  thank you! :raised_hands: